comb_spec_searcher/strategies/rule.py

```diff
@@ -779,25 +734,14 @@ def get_equation(
         lhs_func = get_function(self.comb_class)
         return Eq(lhs_func, self.strategy.get_genf(self.comb_class, funcs))
 
-    def generate_objects_of_size(
-        self, n: int, **parameters: int
-    ) -> Iterator[CombinatorialObjectType]:
-        key = (n,) + tuple(sorted(parameters.items()))
-        res = self.obj_cache.get(key)
-        if res is not None:
-            yield from res
-            return
-        res = []
-        for obj in self.strategy.generate_objects_of_size(
-            self.comb_class, n, **parameters
-        ):
-            yield obj
-            res.append(obj)
-        self.obj_cache[key] = res
-
     def random_sample_object_of_size(
         self, n: int, **parameters: int
     ) -> CombinatorialObjectType:
         return self.strategy.random_sample_object_of_size(
             self.comb_class, n, **parameters
         )
+
+    @staticmethod
+    def sanity_check(n: int) -> bool:
+        # TODO: test more thoroughly
```

Comment:
After implementing the get terms on combinatorial class, we can now check rule.get_terms == rule.comb_class.get_terms